### PR TITLE
More presets

### PR DESCRIPTION
**Pull request by Sollace**
This rewrites the in-game gui with more options and finer controls. (Also centred on the screen. :T)

Big Pony Settings

### CODE REVIEW

File: src/main/java/com/minelittlepony/bigpony/mod/CameraPresets.java [hunk 1]
```diff
@@ -24,6 +26,13 @@ public float getDistance() {
     }
 
     public boolean isEqual(float height, float distance) {
-        return FloatUtils.equals(this.height, height) && FloatUtils.equals(this.distance, distance);
+        return FloatUtils.equals(this.height, height)
+                && FloatUtils.equals(this.distance, distance);
+    }
+
+    public boolean isEqual(float scaleX, float scaleY, float scaleZ) {
+        return this.height == scaleX
```
Comment (mattmess1221) [inline src/main/java/com/minelittlepony/bigpony/mod/CameraPresets.java]: Don't compare floating points with `eq`. Use the above method as an example of what to do.

File: src/main/java/com/minelittlepony/bigpony/mod/gui/GuiBigSettings.java [hunk 2]
```diff
@@ -17,102 +16,86 @@
...
-
-        humanPreset.enabled = !CameraPresets.HUMAN.isEqual(height.getSliderValue(), distance.getSliderValue());
-        marePreset.enabled = !CameraPresets.MARE.isEqual(height.getSliderValue(), distance.getSliderValue());
-        stallionPreset.enabled = !CameraPresets.STALLION.isEqual(height.getSliderValue(), distance.getSliderValue());
-        princessPreset.enabled = !CameraPresets.ALICORN.isEqual(height.getSliderValue(), distance.getSliderValue());
-        fillyPreset.enabled = !CameraPresets.FILLY.isEqual(height.getSliderValue(), distance.getSliderValue());
+        CameraPresets[] values = CameraPresets.values();
+        for (int i = 0; i < cameraPresets.length; i++) {
+            cameraPresets[i].enabled = !values[i].isEqual(height.getSliderValue(), distance.getSliderValue());
+            scalePresets[i].enabled = !values[i].isEqual(xSize.getSliderValue(), ySize.getSliderValue(), zSize.getSliderValue());
```
Comment (mattmess1221) [inline src/main/java/com/minelittlepony/bigpony/mod/gui/GuiBigSettings.java]: Does this work?

Comment (Sollace) [reply src/main/java/com/minelittlepony/bigpony/mod/gui/GuiBigSettings.java]: It wouldn't be in the PR if it didn't.

File: src/main/resources/assets/minebigpony/lang/en_us.lang [hunk 3]
```diff
@@ -0,0 +1,19 @@
+minebp.options.title=Big Pony Settings
```
Comment (mattmess1221) [inline src/main/resources/assets/minebigpony/lang/en_us.lang:1]: I would just call this `bigpony.xxx.xxx`

Comment (mattmess1221) [reply src/main/resources/assets/minebigpony/lang/en_us.lang:1]: Nevermind, leave it. I started it at the first commit.

Comment (Sollace) [reply src/main/resources/assets/minebigpony/lang/en_us.lang:1]: *shrug* I decided to keep it consistent with minelp.

Comment (Sollace) [reply src/main/resources/assets/minebigpony/lang/en_us.lang:1]: Mine Little Pony and Mine Big Pony. Thought that was what you were going for.

File: src/main/java/com/minelittlepony/bigpony/mod/gui/GuiBigSettings.java (hunk 2 above)
Comment (mattmess1221) [reply src/main/java/com/minelittlepony/bigpony/mod/gui/GuiBigSettings.java]: `combinedPresets[i].enabled = cameraPresets[i].enabled || scalePresets[i].enabled;`

**Comment by mattmess1221** [comment]
I think you've misunderstood what the presets do. I had created them so that you can click them to match your camera to what the pony scaling was in minelp. It would make sense if we merge this into the main mod. Then we can actually have it be linked with the models.

**Comment by Sollace** [comment]
Yeah, I kind of got that. It's just I found the giant/ant ones to be sizes I used often whilst playing around with it. Being able to set the body scaling with one click is an added bonus, since it felt weird that you could auto-select camera presets but not bodies.

BigPony could certainly detect a default value from MineLP whene the mod is present. I'd add that as a toggleable option rather than try to merge the two now.

File: src/main/java/com/minelittlepony/bigpony/mod/gui/GuiBigSettings.java (hunk 2 above)
Comment (Sollace) [reply src/main/java/com/minelittlepony/bigpony/mod/gui/GuiBigSettings.java]: @killjoy1221 You pose a very convincing argument. Feature added. :3

**Pull request closed without merging**